classes/01class/exercises/c01-aws02/gpavelar/README.MD

```diff
@@ -0,0 +1,164 @@
+# C01-AWS02
+
+## Commands Execution Output
+
+- Commands to create the S3 Bucket and to copy the file to the bucket:
+
+```bash
+## Creating bucket S3 - first way
+$ aws s3 mb s3://mydevops-bucket
+# output
+make_bucket: mydevops-bucket
+
+## Creating bucket S3 - second way
+$ aws s3api create-bucket --bucket myseconddevops-bucket --region ap-southeast-2
+# output
+/myseconddevops-bucket
+
+## Listing bucket S3
+$ aws s3 ls
+# output
+2020-06-20 13:00:33 mydevops-bucket
+2020-06-20 13:07:04 myseconddevops-bucket
+## or
+$ aws s3api list-buckets
+# output
+BUCKETS 2020-06-20T03:00:33+00:00       mydevops-bucket
+BUCKETS 2020-06-20T03:07:04+00:00       myseconddevops-bucket
+OWNER   gpavelardev     ***********
+
+## copying a file to my s3 bucket.
+aws s3 cp devops-classes/devops-rules.txt s3://mydevops-bucket/devops-classes
+## Another test
+aws s3 cp devops-classes/devops-aws.txt s3://mydevops-bucket/devops-classes/ --grants read=uri=http://acs.amazonaws.com/groups/global/AllUsers
+
+
+```
+
+- Commands to allow the EC2 instance to access the files in S3:
+
+```bash
+## Acessing instance via SSH. I stopped this instance
+$ ssh -A ec2-user@3.104.223.187
+## Trying to list aws s3
+$ aws s3 ls
+# output
+Unable to locate credentials. You can configure credentials by running "aws configure".
+$ aws configure
+## Added my access key id and secret access key with a policy s3bucketreadonly.
+
+## I had already created the user with s3 interface on browser.
+
+## So, I also tried to create a IAM role using AWS CLI
+## Creating role
+aws iam create-role --role-name Test-Role --assume-role-policy-document file://Role-Read-Trust-Policy.json
+
+```
+
+Policy file used:
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": {
+    "Effect": "Allow",
+    "Principal": { "Service": "ec2.amazonaws.com" },
+    "Action": "sts:AssumeRole"
+  }
+}
+```
+
+```bash
+# output
+
+ROLE arn:aws:iam::478433196210:role/Test-Role 2020-06-21T13:44:23+00:00 / AROAW6ZGXHCZNG4L6UYON Test-Role
+ASSUMEROLEPOLICYDOCUMENT 2012-10-17
+STATEMENT sts:AssumeRole Allow
+PRINCIPAL ec2.amazonaws.com
+
+## Attach a managed policy to an IAM role
+
+aws iam attach-role-policy --policy-arn arn:aws:iam::aws:policy/ReadOnlyAccess --role-name Test-Role
```

Comment:
can you create your own policy with the least privileges you need to achieve the exercise objective from the instance?